### PR TITLE
LPS 201218 Unable to add scripted recipients script in workflow

### PR DESCRIPTION
**Pull request by Marina-Gouveia**
Related issue: [LPS-201218](https://liferay.atlassian.net/browse/LPS-201218)
LPP ticket: [LPP-51535](https://liferay.atlassian.net/browse/LPP-51535)

### CODE REVIEW

File: modules/dxp/apps/portal-workflow/portal-workflow-kaleo-designer-web/src/main/resources/META-INF/resources/designer/js/definition-builder/diagram-builder/components/sidebar/sections/notifications/NotificationsInfo.js [hunk 1]
```diff
@@ -373,6 +373,9 @@ const NotificationsInfo = ({
 				else if (recipientType === 'taskAssignees') {
 					recipientDetails = {assignmentType: ['taskAssignees']};
 				}
+				else if (recipientType === 'scriptedRecipient') {
+					recipientDetails = {scriptLanguage: ['groovy']};
+				}
```
Comment (MarinhoFeliphe) [inline modules/dxp/apps/portal-workflow/portal-workflow-kaleo-designer-web/src/main/resources/META-INF/resources/designer/js/definition-builder/diagram-builder/components/sidebar/sections/notifications/NotificationsInfo.js:378]: Hi @Marina-Gouveia, sort this conditional, it should be before [taskAssignees](https://github.com/liferay-workflow/liferay-portal/pull/1013/files#diff-aeacf12a0c6bf7decf8d6bc07c1a7d0e1e6b9800bf3676665664fc379371e830R373).

Thanks!

**Review by MarinhoFeliphe — CHANGES_REQUESTED** [review]

File: modules/dxp/apps/portal-workflow/portal-workflow-kaleo-designer-web/src/main/resources/META-INF/resources/designer/js/definition-builder/diagram-builder/components/sidebar/sections/notifications/NotificationsInfo.js (hunk 1 above)
Comment (Marina-Gouveia) [reply modules/dxp/apps/portal-workflow/portal-workflow-kaleo-designer-web/src/main/resources/META-INF/resources/designer/js/definition-builder/diagram-builder/components/sidebar/sections/notifications/NotificationsInfo.js:375]: Hi @MarinhoFeliphe, done!!

**Review by cgmonte — COMMENTED** [review]
Hi, @Marina-Gouveia! Thanks for working on this! I have tested the fix and I think it is only partially fixed. Although the error message is gone. The "actual result" described in the ticket states that "However, editing the workflow again, the content of the script is gone.", and this is still the case after applying your fix. Is it fixed on your side? See the video:

[Process Builder - Liferay DXP.webm](https://github.com/liferay-workflow/liferay-portal/assets/48384932/4cfc999e-02d5-45b2-828a-7a19cf4c58bd)

**Comment by Marina-Gouveia** [comment]
Hi @cgmonte, This behavior already happens in the master however, the script remains in the Source.

**Comment by cgmonte** [comment]
I may be wrong, but I think this behavior is part of the bug and should fixed, you know? The ticket reads:

Actual Results:
The error "Error:Unable to parse definition" shows up at the left corner. If I switch to the source of the definition before saving, it shows the action is performed successfully. **However, editing the workflow again, the content of the script is gone.**

Was Luiz refering to the Script field in the right sidebar? I know the expected result reads like "The script should be saved successfully.", but that may not describe the fix very well. I can save the workflow without your fix, even though the error appears. What do you think @Luizcarlosqueiroz?

**Comment by Marina-Gouveia** [comment]
Hi @cgmonte,
see https://github.com/liferay-workflow/liferay-portal/pull/1023.

**Pull request closed without merging**